docs/user_guide.md

```diff
@@ -4,63 +4,146 @@ This is a pre-release version of the extension and is not intended for general u
 It may be unstable and documentation is limited.
 If you have any questions, please contact us at [hello@open-diffix.org](mailto:hello@open-diffix.org).
 
-# User Guide
+# Extension configuration
 
-This document provides detailed information about the configuration, behavior and recommended usage
-of __Diffix for PostgreSQL__.
+This document provides detailed information about the configuration, behavior and recommended usage of this extension __pg_diffix__.
 
-## What is Diffix?
+## Background reading
 
-Diffix is a bundled set of mechanisms for anonymizing structured data. It was jointly developed by Aircloak GmbH and
-the Max Planck Institute for Software Systems. Diffix exploits mechanisms that have been in use by national statistics
-offices for decades: aggregation, generalization, noise, suppression, and swapping. It automatically applies these
-mechanisms as needed on a query-by-query basis to minimize noise while ensuring strong anonymity.
+The [Open Diffix website](https://www.open-diffix.org) has general information about the Open Diffix project. The [website FAQ](https://www.open-diffix.org/faq) is a good starting point. [This article](https://www.open-diffix.org/blog/diffix-elm-automates-what-statistics-offices-have-been-doing-for-decades) is a high-level overview of the anonymization mechanisms used by Diffix. [This paper](https://arxiv.org/abs/2201.04351) is a complete specification and privacy analysis of Diffix (version Elm).
 
-## What is Diffix for PostgreSQL?
+## Configuration objects
 
-__Diffix for PostgreSQL__ is an implementation of the Diffix mechanism for PostgreSQL databases.
-It is provided as an extension module for PostgreSQL version 13 or higher.
+Extension behavior is controlled by `security labels` and `settings`. Security labels specific to this extension are assigned to tables, columns, and roles (users). Settings are general parameters associated with system operation. Settings can be assigned in the configuration file and are instantiated per session.
 
-## Configuring the extension
+Only superusers can assign security labels and settings. To remove a security label from an object, set it to `NULL`.
 
-System behaviour can be configured using a combination of custom variables and security labels.
+The command `SELECT * FROM diffix.show_labels();` displays the current security labels assigned to tables and columns by the extension.
 
-### Labeling database objects
+The command `SELECT * FROM diffix.access_level();` displays the current security labels assigned to the role associated with the active session.
```

Comment:
Command can be simplified to `SELECT diffix.access_level()`.

> displays the current security labels assigned to the role associated with the active session.

It's a little more complex than this. It depends on `pg_diffix.session_access_level` which can be overridden by superusers only, by the label assigned to the user, or by `pg_diffix.default_access_level` if no label is assigned to the user.

Should say something like:

"displays the access level of the current user, depending on the security label assigned to the role, or the default configured access level if no label is assigned."

Reply:
Is your point that there are different ways that the security label may have been created, and this does not capture all those ways? But bottom line, there is always a security label associated with the active session right? And this command shows what it is.

Reply:
The confusion came from the word "security label". We have used the term "access level" so far to describe one of `direct | anonymized_trusted | anonymized_untrusted`.

To keep it simple, I would just say "displays returns the access level of the user's current session".

If you're interested in details here is the `get_session_access_level` implementation:

https://github.com/diffix/pg_diffix/blob/1c93e125d2f9002e8f76e21ab2775dd3b2ff32f7/src/auth.c#L62-L89